Core/clim-basic/recording.lisp

```diff
@@ -593,28 +594,28 @@ the associated sheet can be determined."
 
 (defmethod %tree-recompute-extent* ((record compound-output-record))
   ;; Internal helper function
-  (let ((new-x1 0)
-        (new-y1 0)
-        (new-x2 0)
-        (new-y2 0)
-        (first-time t))
-    (map-over-output-records
-     (lambda (child)
-       (unless (null-bounding-rectangle-p child)
-         (if first-time
-             (progn
-               (multiple-value-setq (new-x1 new-y1 new-x2 new-y2)
-                 (bounding-rectangle* child))
-               (setq first-time nil))
-             (with-bounding-rectangle* (cx1 cy1 cx2 cy2) child
-               (minf new-x1 cx1)
-               (minf new-y1 cy1)
-               (maxf new-x2 cx2)
-               (maxf new-y2 cy2)))))
-     record)
-    (if first-time
-        (with-slots (x y) record
-          (values x y x y))
+  (if (zerop (output-record-count record)) ; no children
+      (with-slots (x y) record
+        (values x y x y))
+      (let ((new-x1 0)
+            (new-y1 0)
+            (new-x2 0)
+            (new-y2 0)
+            (first-time t))
+        (flet ((do-child (child)
+                 (cond ((null-bounding-rectangle-p child))
+                       (first-time
```

Comment:
I would find more readable `(unless (null …) (if first-time foo bar))`, but it is opinionated so feel free to ignore this remark.